### ISSUE TITLE
New option for `no-nested-ternary` to allow nested ternary if wrapped in parens

### ISSUE THREAD

**Issue by hafezdivandari** [eslint/eslint#13040]
**What rule do you want to change?**
`no-nested-ternary`

**Does this change cause the rule to produce more or fewer warnings?**
Yes

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option.

**Please provide some example code that this change will affect:**

```js
var thing = foo
     ? bar
     : (baz === qux
        ? quxx
        : foobar);
```

**What does the rule currently do for this code?**
Doesn't allow.

**What will the rule do after it's changed?**
Allows nested ternary if wrapped in parentheses.

**Are you willing to submit a pull request to implement this change?**
Maybe.

**Comment by mdjermanovic** [comment]
Hi @hafezdivandari, thanks for the enhancement proposal!

Would this option also allow more than one level, e.g.:

```js
var thing = foo
    ? bar
    : (baz === qux
       ? quxx
       : (a === b
          ? c
          : d));
```

**Comment by hafezdivandari** [comment]
Hi @mdjermanovic,
Yes I think it should allow multi-level too, it also may be useful to add an option to specify maximum allowed depth.

**Comment by mdjermanovic** [comment]
Thanks for the clarification!

Then, maybe it can be just one `number` option? 

Defaults to `0`, which is the actual behavior.

If set to `1`, allows 1-level nested ternary, but it has to be parenthesized like in the original example.

If set to `2`, allows up to 2 levels deep and they all have to be parenthesized, etc.

**Comment by kaicataldo** [comment]
Do you mind expanding on what problem you're trying to solve with this proposal? I personally feel like a custom rule that enforces that nested ternaries should be wrapped in parens would be better for this.

**Comment by hafezdivandari** [comment]
@kaicataldo sorry, I don't understand what you mean by "custom rule" would please explain more? I think the @mdjermanovic 's solution (adding a new `depth` option to the `no-nested-ternary`) would solve the problem.

**Comment by kaicataldo** [comment]
@hafezdivandari Sorry for the delay here. ESLint is designed in a way that allows anyone to make a custom rule to enforce whatever they want! My feeling in this case is that this would be a better fit for a 3rd party custom rule than a change in core. You can read more about how to write a custom rule [here](https://eslint.org/docs/developer-guide/working-with-plugins).
